deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -0,0 +1,932 @@
+	}
+
+	// Update the fee quoter prices and destinations on the remote chains
+	for _, remoteChain := range c.RemoteChains {
+		out, err = UpdateFeeQuoterDestsChangeset(e, c.updateFeeQuoterDestsConfig(remoteChain))
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run UpdateFeeQuoterDestsChangeset on chain with selector %d: %w", remoteChain.Selector, err)
+		}
+		allProposals = append(allProposals, out.MCMSTimelockProposals...)
+
+		out, err = UpdateFeeQuoterPricesChangeset(e, c.updateFeeQuoterPricesConfig(remoteChain))
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run UpdateFeeQuoterPricesChangeset on chain with selector %d: %w", remoteChain.Selector, err)
+		}
+		allProposals = append(allProposals, out.MCMSTimelockProposals...)
+	}
+
+	// Connect the new chain to the existing chains (use the test router)
+	out, err = ConnectNewChainChangeset.Apply(e, c.connectNewChainConfig())
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run ConnectNewChainChangeset: %w", err)
+	}
+	allProposals = append(allProposals, out.MCMSTimelockProposals...)
+
+	proposal, err := aggregateProposals(
+		e,
+		state,
+		allProposals,
+		nil,
+		fmt.Sprintf("Promote chain with selector %d for testing", c.NewChain.Selector),
+		c.MCMSConfig,
+	)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to build proposal: %w", err)
+	}
+
+	if proposal == nil {
+		return deployment.ChangesetOutput{}, nil
+	}
+	return deployment.ChangesetOutput{MCMSTimelockProposals: []mcmslib.TimelockProposal{*proposal}}, nil
+}
+
+///////////////////////////////////
+// END PromoteNewChainForTestingChangeset
+///////////////////////////////////
+
+///////////////////////////////////
+// START ConnectNewChainChangeset
+///////////////////////////////////
+
+// ConnectionConfig defines how a chain should connect with other chains.
+type ConnectionConfig struct {
+	// RMNVerificationDisabled is true if we do not want the RMN to bless messages FROM this chain.
+	RMNVerificationDisabled bool
+	// AllowListEnabled is true if we want an allowlist to dictate who can send messages TO this chain.
+	AllowListEnabled bool
+}
+
+// ConnectNewChainConfig is a configuration struct for ConnectNewChainChangeset.
+type ConnectNewChainConfig struct {
+	// NewChainSelector is the selector of the new chain to connect.
+	NewChainSelector uint64
+	// NewChainConnectionConfig defines how the new chain should connect with other chains.
+	NewChainConnectionConfig ConnectionConfig
+	// RemoteChains are the chains to connect the new chain to.
+	RemoteChains map[uint64]ConnectionConfig
+	// TestRouter is true if we want to connect via test routers.
+	TestRouter *bool
+	// MCMSConfig is the MCMS configuration, omit to use deployer key only.
+	MCMSConfig *changeset.MCMSConfig
+}
+
+func (c ConnectNewChainConfig) ValidateNewChain(env deployment.Environment, state changeset.CCIPOnChainState) error {
```

Comment:
can it be unexported?

Reply:
https://github.com/smartcontractkit/chainlink/pull/16705/commits/8104b00de8d2820f96e1d73a532a8a32c98f51f0